src/components/inputs/z-input/index.tsx

```diff
@@ -125,13 +125,13 @@ export class ZInput {
     if (!this.isTyping) {
       this.emitStartTyping();
     }
-
+    let validity = this.hostElement.shadowRoot.querySelector("input").validity;
```

Comment:
Not sure the `'input'` selector is enough to query all possible input elements. We should add `textarea`, `select` and `[contenteditable]`